crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -8147,15 +8147,9 @@ impl<'db, 'ast> TypeInferenceBuilder<'db, 'ast> {
             (Type::Union(union_ty), _, _) => union_ty.map(self.db(), |ty| {
                 self.infer_subscript_expression_types(value_node, *ty, slice_ty)
             }),
-            (Type::Intersection(intersection_ty), _, _) => intersection_ty
-                .positive(self.db())
-                .iter()
-                .map(|ty| self.infer_subscript_expression_types(value_node, *ty, slice_ty))
-                .fold(
-                    IntersectionBuilder::new(self.db()),
-                    IntersectionBuilder::add_positive,
-                )
-                .build(),
+            (Type::Intersection(_), _, _) => {
+                todo_type!("Subscript expressions on intersections")
```

Comment:
```suggestion
                // TODO: we can map over the intersection and fold the results back into an intersection,
                // but we need to ignore any errors, which means `infer_subscript_expression_types`
                // needs to return a `Result` rather than eagerly emitting diagnostics.
                todo_type!("Subscript expressions on intersections")
```